keps/sig-api-machinery/20190927-insecure-backend-proxy.md

```diff
@@ -0,0 +1,201 @@
+---
+title: Insecure Backend Proxy
+authors:
+  - "@deads2k"
+owning-sig: sig-xxx
+participating-sigs:
+  - sig-apimachinery
+  - sig-auth
+  - sig-cli
+reviewers:
+  - "@sttts"
+  - "@cheftako"
+  - "@liggitt"
+  - "@soltysh"
+approvers:
+  - "@lavalamp"
+  - "@mikedanese"
+editor: TBD
+creation-date: 2019-09-27
+last-updated: 2019-09-27
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Insecure Backend Proxy
+
+When trying to get logs for a pod, it is possible for a kubelet to have an expired serving certificate.
+If a client chooses, it should be possible to bypass the default behavior of the kube-apiserver and allow the kube-apiserver
+to skip TLS verification of the kubelet to allow gathering logs.  This is especially important for debugging
+misbehaving self-hosted clusters.
```

Comment:
Is this useful in situations outside of self-hosted clusters?

Reply:
Yes, it's conceptually useful anywhere where cert rotation can fail.

If it if wasn't, self-hosted does cover enough of the ecosystem to be widely applicable. Kube-adm and openshift as examples.

Reply:
Is the primary motivation of this change to allow operators to debug why certificate rotation is failing?

> self-hosted does cover enough of the ecosystem to be widely applicable. Kube-adm and openshift as examples.

Does kubelet run self hosted in openshift?

Reply:
> Is the primary motivation of this change to allow operators to debug why certificate rotation is failing?

The proximate cause, yes.  It has come up in the past, but really annoyed me last week.



> Does kubelet run self hosted in openshift?

The kube-controller-manager which signs (or doesn't sign) the CSRs runs on the kubelet as a static pod, so getting logs is important.